tests/NetEscapades.EnumGenerators.Tests/Snapshots/EnumGeneratorTests.CanGenerateEnumExtensionsWithDisplayName.verified.txt

```diff
@@ -0,0 +1,151 @@
+//------------------------------------------------------------------------------
+// <auto-generated>
+//     This code was generated by the NetEscapades.EnumGenerators source generator
+//
+//     Changes to this file may cause incorrect behavior and will be lost if
+//     the code is regenerated.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+#nullable enable
+namespace MyTestNameSpace
+{
+    public static partial class MyEnumExtensions
+    {
+        public static string ToStringFast(this MyTestNameSpace.MyEnum value)
+            => value switch
+            {
+                MyTestNameSpace.MyEnum.First => nameof(MyTestNameSpace.MyEnum.First),
+                MyTestNameSpace.MyEnum.Second => "2nd",
+                MyTestNameSpace.MyEnum.Third => nameof(MyTestNameSpace.MyEnum.Third),
+                MyTestNameSpace.MyEnum.Fourth => "4th",
+                _ => value.ToString(),
+            };
+
+       public static bool IsDefined(MyTestNameSpace.MyEnum value)
+            => value switch
+            {
+                MyTestNameSpace.MyEnum.First => true,
+                MyTestNameSpace.MyEnum.Second => true,
+                MyTestNameSpace.MyEnum.Third => true,
+                MyTestNameSpace.MyEnum.Fourth => true,
+                _ => false,
+            };
+
+        public static bool IsDefined(string name, bool allowMatchingDisplayAttribute = false)
+        {
+            var isDefinedInDisplayAttribute = false;
+            if (allowMatchingDisplayAttribute)
+            {
+                isDefinedInDisplayAttribute = name switch
+                {
+                    "2nd" => true,
+                    "4th" => true,
+                    _ => false,
+                };
+            }
+
+            if (isDefinedInDisplayAttribute)
+            {
+                return true;
+            }
+
+            return name switch
+            {
+                nameof(MyTestNameSpace.MyEnum.First) => true,
+                nameof(MyTestNameSpace.MyEnum.Second) => true,
+                nameof(MyTestNameSpace.MyEnum.Third) => true,
+                nameof(MyTestNameSpace.MyEnum.Fourth) => true,
+                _ => false,
+            };
+        }
+
+        public static bool TryParse(
+#if NETCOREAPP3_0_OR_GREATER
+            [System.Diagnostics.CodeAnalysis.NotNullWhen(true)]
+#endif
+            string? name, 
+            bool ignoreCase, 
+            out MyTestNameSpace.MyEnum value, 
+            bool allowMatchingDisplayAttribute = false)
```

Comment:
I know it's opinionated, but I'd prefer not to use an optional parameter for this methods, and instead to have an additional overload. Which would give us

```
TryParse(string name, out MyEnum value);
TryParse(string name, out MyEnum value, bool ignoreCase);
TryParse(string name, out MyEnum value, bool ignoreCase, bool allowMatchingDisplayAttribute);
```

How does that sound to you?

Reply:
I'll do it, but can you tell me what's the rational behind it?

Reply:
Sure, optional parameters are defined in the callee's code rather than in the library itself, which _can_ lead to issues with library maintenance. I don't forsee it being much of an issue in this case, but I've been bitten about it before. [This sonarcube rule covers it a bit too](https://rules.sonarsource.com/csharp/RSPEC-2360).

Reply:
Done